packages/medusa/src/scripts/discount-rule-migration.ts

```diff
@@ -0,0 +1,132 @@
+import dotenv from "dotenv"
+import { createConnection } from "typeorm"
+import Logger from "../loaders/logger"
+import {
+  DiscountCondition,
+  DiscountConditionOperator,
+  DiscountConditionType,
+} from "../models/discount-condition"
+import { DiscountConditionProduct } from "../models/discount-condition-product"
+import { DiscountRule } from "../models/discount-rule"
+dotenv.config()
+
+const typeormConfig = {
+  type: process.env.TYPEORM_CONNECTION,
+  url: process.env.TYPEORM_URL,
+  username: process.env.TYPEORM_USERNAME,
+  password: process.env.TYPEORM_PASSWORD,
+  database: process.env.TYPEORM_DATABASE,
+  migrations: [process.env.TYPEORM_MIGRATIONS as string],
+  entities: [process.env.TYPEORM_ENTITIES],
+  logging: true,
+}
+
+const migrate = async function ({ typeormConfig }): Promise<void> {
+  const connection = await createConnection(typeormConfig)
+
+  const BATCH_SIZE = 1000
+
+  await connection.transaction(async (manager) => {
+    const discountRuleCount = await manager
+      .createQueryBuilder()
+      .from(DiscountRule, "dr")
+      .getCount()
+
+    let offset = 0
+    while (offset < discountRuleCount) {
+      const discountRules = await manager
+        .createQueryBuilder()
+        .from(DiscountRule, "dr")
+        .select("dr.id", "dr_id")
+        .innerJoin(
+          "discount_rule_product",
+          "drp",
+          "dr_id = drp.discount_rule_id"
```

Comment:
issue: The alias is strangely not being picked up here. We can just remove it and use the id directly
```suggestion
          "dr.id = drp.discount_rule_id"
```